testsuite/MDAnalysisTests/visualization/test_streamlines.py

```diff
@@ -0,0 +1,77 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+from __future__ import absolute_import, print_function
+import numpy as np
+import MDAnalysis
+from MDAnalysis.visualization import streamlines
+from MDAnalysis.coordinates.XTC import XTCWriter
+from MDAnalysisTests.datafiles import Martini_membrane_gro
+import pytest
+import matplotlib
+matplotlib.use('agg')
+import matplotlib.pyplot as plt
+import os
+
+def test_streamplot_2D(tmpdir):
+    # simple roundtrip test to ensure that
+    # a plot is generated by the 2D streamplot
+    # code
+    u = MDAnalysis.Universe(Martini_membrane_gro)
```

Comment:
The fake trajectory should become a session-level fixture because it is used by the 2D and 3D stream line tests
```python
@pytest.fixture(scope="session")
def membrane_xtc(tmpdir_factory):
    u = MDAnalysis.Universe(Martini_membrane_gro)

    x_delta, y_delta  = 0.5, 0.3
    tmp_xtc = tmpdir_factory.mktemp('streamlines').join('dummy.xtc')
    with  XTCWriter(tmp_xtc, n_atoms=u.atoms.n_atoms) as xtc_writer:
        for i in range(5):
           u.atoms.translate([x_delta, y_delta, 0])
           xtc_writer.write(u.atoms)
           x_delta += 0.1
           y_delta += 0.08
    return tmp_xtc

def test_streamplot_2D(membrane_xtc, tmpdir):
   ...

def test_streamplot_3D(membrane_xtc, tmpdir):
   ...
```

You could put the session-level fixture into a separate file and import it but it seems easier to combine both the tests for 2D and 3D into a single `test_streamlines.py`.

Reply:
Note that one has to use the [`tmpdir_factory`](https://docs.pytest.org/en/latest/tmpdir.html#the-tmpdir-factory-fixture) for a session-scoped fixture; the `tmpdir` fixture is function-scoped and one cannot mix the two scopes.